lib/elixir_bench/benchmarks/benchmarks.ex

```diff
@@ -51,12 +51,11 @@ defmodule ElixirBench.Benchmarks do
     Repo.fetch(where(Job, uuid: ^uuid))
   end
 
-  def fetch_running_job_by_reference(branch_name, commit_sha) do
+  defp fetch_job_by_repo_and_commit(repo, commit_sha) do
     query =
       Job
-      |> last
-      |> Job.unfinished()
-      |> where(branch_name: ^branch_name, commit_sha: ^commit_sha)
+      |> Job.filter_by_repo(repo.id)
+      |> where(commit_sha: ^commit_sha)
```

Comment:
as we removed `last` won't this return multiple jobs now instead of just one? I think the code still works but it's confusing and we potentially do more than we should

Reply:
you are right :100:... Repo.fetch will break if more than one Job is found... I added a test case for that and put back the `last`

Reply:
had to do a little change in tests as we just check jobs for a given repo and commit-sha, ignoring the branch name

Reply:
Repo.fetch wouldn't fail - `Repo.one` would fail :)

Reply:
`Repo.fetch` calls one in the end, see `ElixirBench.Repo`, I tested and it breaks :male_detective: